samples/snippets/src/main/java/com/example/dialogflow/DetectIntentWithLocation.java

```diff
@@ -0,0 +1,79 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.example.dialogflow;
+
+// [START dialogflow_detect_intent_with_location]
+
+import com.google.api.gax.rpc.ApiException;
+import com.google.cloud.dialogflow.v2beta1.DetectIntentResponse;
+import com.google.cloud.dialogflow.v2beta1.QueryInput;
+import com.google.cloud.dialogflow.v2beta1.QueryResult;
+import com.google.cloud.dialogflow.v2beta1.SessionName;
+import com.google.cloud.dialogflow.v2beta1.SessionsClient;
+import com.google.cloud.dialogflow.v2beta1.SessionsSettings;
+import com.google.cloud.dialogflow.v2beta1.TextInput;
+import com.google.common.collect.Maps;
+import java.io.IOException;
+import java.util.List;
+import java.util.Map;
+
+public class DetectIntentWithLocation {
+
+  // DialogFlow API Detect Intent sample with text inputs.
+  public static Map<String, QueryResult> detectIntentWithLocation(
+      String projectId, String locationId, List<String> texts, String sessionId,
+      String languageCode)
+      throws IOException, ApiException {
+    SessionsSettings sessionsSettings = SessionsSettings.newBuilder()
+        .setEndpoint(locationId + "-dialogflow.googleapis.com:443").build();
+    Map<String, QueryResult> queryResults = Maps.newHashMap();
+    // Instantiates a client
+    try (SessionsClient sessionsClient = SessionsClient.create(sessionsSettings)) {
+      // Set the session name using the projectId (my-project-id), locationId and sessionId (UUID)
+      SessionName session = SessionName
+          .ofProjectLocationSessionName(projectId, locationId, sessionId);
+      System.out.println("Session Path: " + session.toString());
+
+      // Detect intents for each text input
+      for (String text : texts) {
+        // Set the text (hello) and language code (en-US) for the query
+        TextInput.Builder textInput =
+            TextInput.newBuilder().setText(text).setLanguageCode(languageCode);
+
+        // Build the query with the TextInput
+        QueryInput queryInput = QueryInput.newBuilder().setText(textInput).build();
+
+        // Performs the detect intent request
+        DetectIntentResponse response = sessionsClient.detectIntent(session, queryInput);
+
+        // Display the query result
+        QueryResult queryResult = response.getQueryResult();
+
+        System.out.println("====================");
+        System.out.format("Query Text: '%s'\n", queryResult.getQueryText());
+        System.out.format(
+            "Detected Intent: %s (confidence: %f)\n",
+            queryResult.getIntent().getDisplayName(), queryResult.getIntentDetectionConfidence());
+        System.out.format("Fulfillment Text: '%s'\n", queryResult.getFulfillmentText());
+
+        queryResults.put(text, queryResult);
+      }
+    }
+    return queryResults;
+  }
+}
+// [END dialogflow_detect_intent_text]
```

Comment:
This one needs to match the above snippet of too

Reply:
done.